File: app/db/models/staff.py
from datetime import datetime

from sqlalchemy import String, Text
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.db.base import Base
from app.db.models.resource_staff import resource_staff


class Staff(Base):
    __tablename__ = "staff"

    id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
    first_name: Mapped[str] = mapped_column(String(255))
    last_name: Mapped[str] = mapped_column(String(255))
    clinical_level: Mapped[str] = mapped_column(String(20))
    notes: Mapped[str | None] = mapped_column(Text, nullable=True)
    created_at: Mapped[datetime | None] = mapped_column(nullable=True)
    updated_at: Mapped[datetime | None] = mapped_column(nullable=True)

    resources: Mapped[list[Resource]] = relationship(
        secondary=resource_staff, back_populates="staff"
    )

    @property
    def full_name(self) -> str:
        return f"{self.first_name} {self.last_name}"

File: app/Models/Staff.php
<?php

namespace App\Models;

use App\Enums\ClinicalLevel;
use Database\Factories\StaffFactory;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsToMany;

class Staff extends Model
{
    /** @use HasFactory<StaffFactory> */
    use HasFactory;

    protected $fillable = [
        'first_name',
        'last_name',
        'clinical_level',
        'notes',
    ];

    protected function casts(): array
    {
        return [
            'clinical_level' => ClinicalLevel::class,
        ];
    }

    public function getFullNameAttribute(): string
    {
        return "{$this->first_name} {$this->last_name}";
    }

    public function resources(): BelongsToMany
    {
        return $this->belongsToMany(Resource::class, 'resource_staff')
            ->withTimestamps();
    }
}
